amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -490,10 +518,17 @@ public void onCharacteristicRead(BluetoothGatt gatt,
                                                  BluetoothGattCharacteristic characteristic,
                                                  int status) {
                     mRWinProgress = false;
-                    byte[] responseBytes = characteristic.getValue();
-                    Log.d(TAG, "->->-> onCharacteristicRead status: " + (status == 0 ? "Success. " : status)
-                            + bytesToHexString(responseBytes));
                     if (status == BluetoothGatt.GATT_SUCCESS) {
+                        // On the first successful read we enable the services
+                        if (mBleConnectionState == BleConnectionState.BLE_CONNECTED) {
+                            Log.d(TAG, "GATT services initialized");
+                            mBleConnectionState = AmazonFreeRTOSConstants.BleConnectionState.BLE_INITIALIZED;
+                            mBleConnectionStatusCallback.onBleConnectionStatusChanged(mBleConnectionState);
```

Comment:
I think the BLE_INITIALIZED callback should be invoked only after initialization step is completed, probably in `onMtuChanged` callback as it was done before.

Reply:
Will that call back always be invoked?